localstack/aws/connect.py

```diff
@@ -0,0 +1,473 @@
+    from mypy_boto3_ec2 import EC2Client
+    from mypy_boto3_ecr import ECRClient
+    from mypy_boto3_es import ElasticsearchServiceClient
+    from mypy_boto3_events import EventBridgeClient
+    from mypy_boto3_firehose import FirehoseClient
+    from mypy_boto3_iam import IAMClient
+    from mypy_boto3_kinesis import KinesisClient
+    from mypy_boto3_kms import KMSClient
+    from mypy_boto3_lambda import LambdaClient
+    from mypy_boto3_logs import CloudWatchLogsClient
+    from mypy_boto3_opensearch import OpenSearchServiceClient
+    from mypy_boto3_redshift import RedshiftClient
+    from mypy_boto3_resource_groups import ResourceGroupsClient
+    from mypy_boto3_resourcegroupstaggingapi import ResourceGroupsTaggingAPIClient
+    from mypy_boto3_route53 import Route53Client
+    from mypy_boto3_route53resolver import Route53ResolverClient
+    from mypy_boto3_s3 import S3Client
+    from mypy_boto3_s3control import S3ControlClient
+    from mypy_boto3_secretsmanager import SecretsManagerClient
+    from mypy_boto3_ses import SESClient
+    from mypy_boto3_sns import SNSClient
+    from mypy_boto3_sqs import SQSClient
+    from mypy_boto3_ssm import SSMClient
+    from mypy_boto3_stepfunctions import SFNClient
+    from mypy_boto3_sts import STSClient
+    from mypy_boto3_transcribe import TranscribeClient
+
+
+LOG = logging.getLogger(__name__)
+
+
+def attribute_name_to_service_name(attribute_name):
+    """
+    Converts a python-compatible attribute name to the boto service name
+    :param attribute_name: Python compatible attribute name. In essential the service name, if it is a python keyword
+        prefixed by `aws`, and all `-` replaced by `_`.
+    :return:
+    """
+    if attribute_name.startswith("aws"):
+        # remove aws prefix for services named like a keyword.
+        # Most notably, "awslambda" -> "lambda"
+        attribute_name = attribute_name[3:]
+    # replace all _ with -: cognito_idp -> cognito-idp
+    return attribute_name.replace("_", "-")
+
+
+#
+# Data transfer object
+#
+
+INTERNAL_REQUEST_PARAMS_HEADER = "x-localstack-data"
+"""Request header which contains the data transfer object."""
+
+
+class InternalRequestParameters(TypedDict):
+    """
+    LocalStack Data Transfer Object.
+
+    This is sent with every internal request and contains any additional information
+    LocalStack might need for the purpose of policy enforcement. It is serialised
+    into text and sent in the request header.
+
+    Attributes can be added as needed. The keys should roughly correspond to:
+    https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_condition-keys.html
+    """
+
+    source_arn: str | None
+    """ARN of resource which is triggering the call"""
+
+    service_principal: str | None
+    """Service principal making this call"""
+
+
+def dump_dto(data: InternalRequestParameters) -> str:
+    # TODO@viren: Minification can be improved using custom JSONEncoder that uses shortened keys
+
```

Comment:
I don't really see a point in adding the TODO here (especially not with a name reference).
It should either be implemented right away, or (what I would suppose) it will come up if it really has an impact (which I doubt) and will then be tackled in the next iteration.

Reply:
Yes, this was an overlook. I also doubt it will be necessary, but it might be if the headers become too large. Will convert it to a general comment, instead of a todo.